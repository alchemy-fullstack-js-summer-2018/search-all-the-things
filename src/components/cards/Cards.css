:local(.cards) {
  display: grid;
  grid-template: repeat(3, 1fr) / repeat(6, 1fr);
  padding: 0;
  border-left: 1px solid lightgrey;
  li {
    margin: 5px;
    text-align: center;
  }

  img {
    width: 100%;
  }
}