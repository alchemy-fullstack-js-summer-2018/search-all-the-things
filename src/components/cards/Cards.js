import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Card from './Card.js';
import styles from './Cards.css';
export default class Cards extends Component {

  static propTypes = {
    cards: PropTypes.arrayOf(Object)
  };

  render() {
    const { cards } = this.props;

    return (
      <ul className={styles.cards}>
        {cards.map((card, i) => (
          <Card key={i} card={card}/>
        ))}
      </ul>
    );
    
  }
}