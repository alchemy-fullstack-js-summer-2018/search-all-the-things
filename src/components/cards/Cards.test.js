import React from 'react';
import  { shallow } from 'enzyme';
import Cards from './Cards';
import toJSON from 'enzyme-to-json';

const cards = [
  { id: 1 },
  { id: 2 },
  { id: 3 }
];

describe('Cards', () => {

  it('renders correctly', () => {
    const wrapper = shallow(<Cards cards={cards}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});