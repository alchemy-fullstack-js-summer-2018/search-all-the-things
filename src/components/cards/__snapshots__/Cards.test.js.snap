// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards renders correctly 1`] = `
<ul>
  <Card
    card={
      Object {
        "id": 1,
      }
    }
    key="0"
  />
  <Card
    card={
      Object {
        "id": 2,
      }
    }
    key="1"
  />
  <Card
    card={
      Object {
        "id": 3,
      }
    }
    key="2"
  />
</ul>
`;
